apps/arweave/src/ar_node_worker.erl

```diff
@@ -314,15 +314,15 @@ move_tx_to_mining_pool(StateIn, TX, GS) ->
 	} = StateIn,
 	{NewGS, _} = ar_gossip:send(GS, {move_tx_to_mining_pool, TX}),
 	{ok, [
-		{txs, TXs ++ [TX]},
+		{txs, sets:add_element(TX, TXs)},
 		{gossip, NewGS},
-		{waiting_txs, WaitingTXs -- [TX]}
+		{waiting_txs, sets:del_element(TX, WaitingTXs)}
 	]}.
 
 drop_waiting_txs(#{ waiting_txs := WaitingTXs }, DropTXs, GS) ->
 	{NewGS, _} = ar_gossip:send(GS, {drop_waiting_txs, DropTXs}),
 	DropSet = sets:from_list(lists:map(fun(TX) -> TX#tx.id end, DropTXs)),
-	FilteredTXs = lists:filter(
+	FilteredTXs = sets:filter(
```

Comment:
We could simplify it as `sets:subtract(WaitingTXs, sets:from_list(DropTXs))`

Reply:
Done.